File: examples/openclaw_agent.py
from synapse.models.agent import AgentDefinition, AgentKind
from synapse.sdk import SynapseClient


def main() -> None:
    client = SynapseClient()
    browser = client.browser

    client.register_agent(
        AgentDefinition(
            agent_id="openclaw-example",
            kind=AgentKind.OPENCLAW,
            name="OpenClaw Example",
            description="Example OpenClaw-style Synapse SDK agent.",
        )
    )

    page = browser.open("https://example.com")
    extracted = browser.extract("h1")
    tool_result = browser.call_tool("web.search", {"query": "Synapse browser runtime"})
    browser.send_agent_message(
        sender_agent_id="openclaw-example",
        recipient_agent_id="codex-example",
        content="OpenClaw finished initial discovery.",
        metadata={"page_title": page.page.title},
    )

    print({"title": page.page.title, "extract": extracted.model_dump(), "tool": tool_result})
    client.close()


if __name__ == "__main__":
    main()

File: sdk/javascript/examples/openclaw-agent.mjs
import { SynapseClient, createAgentDefinition } from "../src/index.js";

const agentId = "openclaw-js-example";
const client = new SynapseClient({
  baseUrl: "http://127.0.0.1:8000",
  agentId
});

await client.registerAgent(
  createAgentDefinition({
    agentId,
    kind: "openclaw",
    name: "OpenClaw JS Example",
    description: "Example OpenClaw-style agent using the JavaScript SDK.",
    allowedDomains: ["example.com"],
    allowedTools: ["web.search"]
  })
);

const browser = client.browser;
const page = await browser.open("https://example.com");
const heading = await browser.extract("h1");
const toolResult = await browser.callTool("web.search", {
  query: "Synapse browser runtime"
});

console.log({
  title: page.page.title,
  headingCount: heading.matches.length,
  toolResult
});
